extensions/amp-date-picker/1.0/component/date-range-picker.tsx

```diff
@@ -0,0 +1,435 @@
+  ]);
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      clear,
+      setDates: (startDate: Date, endDate: Date) =>
+        selectDateRange({from: startDate, to: endDate}),
+      startToday: startDateInput.setToToday,
+      endToday: endDateInput.setToToday,
+    }),
+    [clear, selectDateRange, startDateInput, endDateInput]
+  );
+
+  /**
+   * For inputs that are valid dates, update the date-picker value.
+   */
+  const handleStartInput = useCallback(
+    (e: InputEvent) => {
+      const {target} = e;
+      if ((target as HTMLInputElement).type === 'hidden') {
+        return;
+      }
+
+      const date = parseDate((target as HTMLInputElement).value);
+      if (date && isValid(date)) {
+        startDateInput.setDate(date);
+      }
+    },
+    [startDateInput, parseDate]
+  );
+
+  /**
+   * For inputs that are valid dates, update the date-picker value.
+   */
+  const handleEndInput = useCallback(
+    (e: InputEvent) => {
+      const {target} = e;
+      if ((target as HTMLInputElement).type === 'hidden') {
+        return;
+      }
+
+      const date = parseDate((target as HTMLInputElement).value);
+      if (date && isValid(date)) {
+        endDateInput.setDate(date);
+      }
+    },
+    [endDateInput, parseDate]
+  );
+
+  useEffect(() => {
+    startDateInput.initialize(containerRef.current!);
+    endDateInput.initialize(containerRef.current!);
+
+    // This should only be called on first render
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    // Since we are passing this ref directly into the container, we can assume
+    // that containerRef.current is defined in this case
+    const containerEl = containerRef.current!;
+    const document = containerEl.ownerDocument;
+    const startInputEl = startDateInput.ref.current;
+    const endInputEl = endDateInput.ref.current;
+    if (!document) {
+      return;
+    }
+
+    const handleFocus = (event: FocusEvent) => {
+      if (event.target === startInputEl) {
+        setFocusedInput('start-input');
+      }
+      if (event.target === endInputEl) {
+        setFocusedInput('end-input');
+      }
+      transitionTo('overlay-open-input');
+    };
+
+    const handleDocumentKeydown = (event: KeyboardEvent) => {
+      if (event.key === Keys_Enum.ESCAPE) {
+        transitionTo('overlay-closed');
+      }
+    };
+    const handleInputKeydown = (event: KeyboardEvent) => {
+      const {target} = event;
+      if ((target as HTMLInputElement).type === 'hidden') {
+        return;
+      }
+
+      if (event.key === Keys_Enum.DOWN_ARROW) {
+        transitionTo('overlay-open-picker');
+      }
+    };
+    const handleClick = (event: MouseEvent) => {
+      const clickWasInDatePicker =
+        event.target === startInputEl ||
+        event.target === endInputEl ||
+        containerEl.contains(event.target as Node);
+      if (!clickWasInDatePicker) {
+        transitionTo('overlay-closed');
+      }
```

Comment:
This is not working as expected when the calendar component is in the shadow DOM. 359 is returning false when the user clicks inside the calendar.